docs/spark/how-to-guides/ubuntu-instructions.md

```diff
@@ -0,0 +1,244 @@
+     - Run the following command:
+
+       ```bash
+       mkdir -p ~/bin/maven
+       cd ~/bin/maven
+       wget https://www-us.apache.org/dist/maven/maven-3/3.6.0/binaries/apache-maven-3.6.0-bin.tar.gz
+       tar -xvzf apache-maven-3.6.0-bin.tar.gz
+       ln -s apache-maven-3.6.0 current
+       export M2_HOME=~/bin/maven/current
+       export PATH=${M2_HOME}/bin:${PATH}
+       source ~/.bashrc
+       ```
+       
+       Note that these environment variables will be lost when you close your terminal. If you want the changes to be permanent, add the `export` lines to your `~/.bashrc` file.
+     - Verify you are able to run `mvn` from your command-line
+       <details>
+       <summary>&#x1F4D9; Click to see sample mvn -version output</summary>
+       
+       ```
+       Apache Maven 3.6.0 (97c98ec64a1fdfee7767ce5ffb20918da4f719f3; 2018-10-24T18:41:47Z)
+       Maven home: ~/bin/apache-maven-3.6.0
+       Java version: 1.8.0_191, vendor: Oracle Corporation, runtime: /usr/lib/jvm/java-8-openjdk-amd64/jre
+       Default locale: en, platform encoding: UTF-8
+       OS name: "linux", version: "4.4.0-17763-microsoft", arch: "amd64", family: "unix"
+       ```
+
+  4. Install **[Apache Spark 2.3+](https://spark.apache.org/downloads.html)**
+     - Download [Apache Spark 2.3+](https://spark.apache.org/downloads.html) and extract it into a local folder (e.g., `~/bin/spark-2.3.2-bin-hadoop2.7`)
+     - Add the necessary [environment variables](https://www.java.com/en/download/help/path.xml) `SPARK_HOME` e.g., `~/bin/spark-2.3.2-bin-hadoop2.7/`
+
+       ```bash
+       export SPARK_HOME=~/bin/spark-2.3.2-hadoop2.7
+       export PATH="$SPARK_HOME/bin:$PATH"
+       source ~/.bashrc
+       ```
+       
+       Note that these environment variables will be lost when you close your terminal. If you want the changes to be permanent, add the `export` lines to your `~/.bashrc` file.
+     - Verify you are able to run `spark-shell` from your command-line
+        <details>
+        <summary>&#x1F4D9; Click to see sample console output</summary>
+        
+        ```
+        Welcome to
+              ____              __
+             / __/__  ___ _____/ /__
+            _\ \/ _ \/ _ `/ __/  '_/
+           /___/ .__/\_,_/_/ /_/\_\   version 2.3.2
+              /_/
+
+        Using Scala version 2.11.8 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_201)
+        Type in expressions to have them evaluated.
+        Type :help for more information.
+
+        scala> sc
+        res0: org.apache.spark.SparkContext = org.apache.spark.SparkContext@6eaa6b0c
+        ```
+                   
+        </details>
+
+Please make sure you are able to run `dotnet`, `java`, `mvn`, `spark-shell` from your command-line before you move to the next section. Feel there is a better way? Please [open an issue](https://github.com/dotnet/spark/issues) and feel free to contribute.
+
+## Building
```

Comment:
This change will break the links to this heading, the links should be fixed too.

```suggestion
## Build
```